packages/app/features/checks/utils/useSendCheckData.ts

```diff
@@ -0,0 +1,24 @@
+import { type UseQueryResult, useQuery } from '@tanstack/react-query'
+import { baseMainnetClient, sendCheckAbi, sendCheckAddress } from '@my/wagmi'
+import { type Hex, getContract } from 'viem'
+import type { SendCheckData } from 'app/features/checks/types'
+
+export const useSendCheckData = (
+  ephemeralAddress: Hex,
+  queryProps?
+): UseQueryResult<SendCheckData> => {
+  return useQuery({
+    queryKey: ['sendCheckData'],
```

Comment:
you want it to update if the ephemeralAddress changes
```suggestion
    queryKey: ['sendCheckData', ephemeralAddress],
```